tests/nada-tests/src/shuffle.py

```diff
@@ -2,45 +2,105 @@
 
 from typing import List
 
-from nada_dsl import Output, PublicInteger, SecretInteger
+from nada_dsl import Integer, Output, PublicInteger, SecretInteger
 
+import numpy as np
 import nada_numpy as na
-# Step 0: Nada Numpy is imported with this line
 from nada_numpy import shuffle
 
+def bool_to_int(bool):
+    """Casting bool to int"""
+    return bool.if_else(Integer(0), Integer(1))
+
+def count(vec, element):
+    """
+    Counts the number of times element is in vec.
+    """
+
+    result = Integer(0)
+    for e in vec:
+        b = ~(element == e)
+        int_b = bool_to_int(b)
+        result += int_b
+
+    return result
+    
 
 def nada_main() -> List[Output]:
 
+    n = 8
+
     parties = na.parties(2)
-    a = na.array([8], parties[0], "A", na.Rational)
-    b = na.array([8], parties[0], "B", na.SecretRational)
-    c = na.array([8], parties[0], "C", PublicInteger)
-    d = na.array([8], parties[0], "D", SecretInteger)
+    a = na.array([n], parties[0], "A", na.Rational)
+    b = na.array([n], parties[0], "B", na.SecretRational)
+    c = na.array([n], parties[0], "C", PublicInteger)
+    d = na.array([n], parties[0], "D", SecretInteger)
 
+    
     # As a function
 
     shuffled_a = shuffle(a)
     shuffled_b = shuffle(b)
     shuffled_c = shuffle(c)
     shuffled_d = shuffle(d)
 
+    # 1. Show shuffle works for Rational, SecretRational, PublicInteger and SecretInteger
     result_a = shuffled_a - shuffled_a
     result_b = shuffled_b - shuffled_b
     result_c = shuffled_c - shuffled_c
     result_d = shuffled_d - shuffled_d
 
+    # 2. Randomness: show at least one element is in a different position 
+    # true if equal
+    diff_position_bool = [a[i] == shuffled_a[i] for i in range(n)] 
+    # cast to int (true -> 0 and false -> 1)
+    diff_position = np.array([bool_to_int(element) for element in diff_position_bool]) 
+    # add them
+    sum = diff_position.sum()
+    # if all are equal => all are 0 => sum is zero
+    at_least_one_diff_element = sum > Integer(0)
+    
+    # 3. Show elements are preserved: 
+    check = Integer(0)
+    for ai in a:
+        nr_ai_in_shufled_a = count(shuffled_a, ai)
+        nr_ai_in_a = count(a, ai)
+        check += bool_to_int(nr_ai_in_shufled_a == nr_ai_in_a)
+    elements_are_preserved = check == Integer(0)
```

Comment:
This is `n^2`, if we have `nlogn` sorting it'd make sense to sort both and check them one by one. Otherwise, this is also good -- it's meant to be just an example.

Reply:
That is true. Although you can have repetitions. So, if you have repeated numbers you have to count them anyway and check that the number of elements match.

Just noticed that with the new way of testing we can do all this in the clear. I will merge this anyway.